src/fragments/lib/auth/common/mfa/flows.mdx

```diff
@@ -0,0 +1,318 @@
+ 
+Once the authenticator app is set up, the user can generate a TOTP code and provide it to the library to complete the sign in process.
+
+import iosSignIn from "/src/fragments/lib/auth/ios/totp/sign_in.mdx";
+
+<Fragments fragments={{ios: iosSignIn}} />
+
+The TOTP code can be obtained from the user via a text field or any other means. Once the user provides the TOTP code, call `confirmSignIn` with the TOTP code as the `challengeResponse` parameter.
+
+import iosConfirmSignIn from "/src/fragments/lib/auth/ios/totp/confirm_sign_in.mdx";
+
+<Fragments fragments={{ios: iosConfirmSignIn}} />
+
+### Enabling TOTP after a user is signed in
+
+TOTP MFA can be set up after a user has signed in. This can be done when the following conditions are met:
+
+- MFA is marked as **Optional** or **Required** in Cognito User Pool.
+- TOTP is marked as an enabled MFA method in Cognito user pool
+
+TOTP setup can be achieved by calling `setUpTOTP` and `verifyTOTPSetup` APIs that are available in the `Auth` category.
+
+Invoke the `setUpTOTP` API to generate a `TOTPSetupDetails` object that contains the `sharedSecret` which will be used to either to generate a QR code or can be manually entered into an Authenticator app.
+
+**`setUpTOTP` API**
+
+import iosSetUpTOTP from "/src/fragments/lib/auth/ios/totp/set_up_totp.mdx";
+
+<Fragments fragments={{ios: iosSetUpTOTP}} />
+
+Once the authenticator app is set up, the user can get a TOTP code that is generated by the Authenticator app and can provide it to the library to using the `verifyTOTPSetup` API to complete the TOTP setup process.
+
+**`verifyTOTPSetup` API**
+
+import iosVerifyTOTPSetup from "/src/fragments/lib/auth/ios/totp/verify_totp_setup.mdx";
+
+<Fragments fragments={{ios: iosVerifyTOTPSetup}} />
+
+### Recovering from a lost TOTP device
+
+If a user loses access to their TOTP device, they would need to contact an administrator to help get access to their account. Based on the Cognito User Pool configuration, the administrator can use the [AdminSetUserMFAPreference](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminSetUserMFAPreference.html) to either change the MFA preference to a different MFA method or to disable MFA for the user.
+
+<Callout warning>
+
+In a scenario where MFA is marked as **Required** in Cognito User Pool and another MFA method is not set up, the administrator would need to first initiate an [AdminUpdateUserAttributes](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminUpdateUserAttributes.html) call and update the user’s phone number attribute. Once this is complete, the administrator can continue changing the MFA preference to SMS as suggested above.
+
+</Callout>
+
+
+## MFA Preferences
```

Comment:
please change this title to "Setting up user preferred MFA option"

Reply:
Done.